src/celengine/body.cpp

```diff
@@ -746,6 +745,44 @@ Vector3d Body::planetocentricToCartesian(const Vector3d& lonLatAlt) const
 }
 
 
+/*! Convert planetocentric coordinates to geodetic ones.
+ *
+ * Formulae are taken from DOI 10.1007/s00190-011-0514-7.
+ *
+ * @param lon longitude
+ * @param lat latitude
+ * @param alt altitude (height above the surface)
+ *
+ * @return geodetic coordinates
+ */
+Vector3d Body::planetocentricToGeodetic(double lon, double lat, double alt) const
```

Comment:
I think this name is confusing - the calculation being done here appears to be the geodetic-to-Cartesian transformation.